test/tla/specs/checker/Solver.tla

```diff
@@ -0,0 +1,54 @@
+---------------------------- MODULE Solver ------------------------------------
+(*
+ * An abstract model of a constraint solver:
+ *
+ * 1. The states are represented with integers.
+ *
+ * 2. The constraints are represented with sets of integers (predicates) and
+ * sets of pairs of integers (relations).
+ *
+ * Igor Konnov, Informal Systems, 2022
+ *)
+
+EXTENDS Integers, Sequences, Solver_typedefs
+
+CONSTANTS
+    \* The set of all potential states.
+    \*
+    \* @type: Set(Int);
+    STATES
+
+(*
+ * context.unary contains constraints over states that are layered by time
+ * frames.  Every set S of integers in unary is understood as a predicate
+ * that evaluates to true on the elements of the set.
```

Comment:
```suggestion
 * context.unary contains constraints over states that are indexed by time
 * frames. Every set S of integers in `unary` is understood as a predicate
 * that evaluates to true on the elements of the set.
```

Reply:
This should also explain what a set of sets of ints in `unary` represents.
One has to read the definition of `IsModel` to understand.